archive/oxford_invariants/2021summer/week3_p1.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import algebra.big_operators.order
+import algebra.big_operators.ring
+import data.nat.modeq
+
+/-!
+# The Oxford Invariants Puzzle Challenges - Summer 2021, Week 3, Problem 1
+
+## Original statement
+
+Let `n ≥ 3`, `a₁, ..., aₙ` strictly positive integers such that `aᵢ ∣ aᵢ₋₁ + aᵢ₊₁` for
+`i = 2, ..., n - 1`. Show that `∑ᵢ₌₁ⁿ⁻¹ a₁aₙ/aᵢaᵢ₊₁ ∈ ℕ`.
```

Comment:
I think I would write out the sum in ordinary LaTeX. The unicode subscripts and superscripts on the summation don't really help me.

Reply:
But will it even ever be rendered?

Reply:
I think Bryans Observable notebooks will, and some doctools also render latex, I thought.

Reply:
doc-gen will render latex. See https://leanprover-community.github.io/mathlib_docs/data/mv_polynomial/basic.html for example

Reply:
Yes, sure. But docgen doesn't go through `archive`, right?

Reply:
So what I should put in Latex? Just the sums? Every formula that's not in Lean? I find it pretty readable already and putting it in Latex will worsen plain text readability.

Reply:
I agree the unicode subscripts and superscripts are less readable than (un-rendered) latex.

Reply:
I changed all the places where sums appeared to Latex. I really doubt this is an improvement...

Reply:
You're right. I wasn't anticipating you would remove all the unicode, just the weird compound super/subscripts on the sigmas.

Reply:
So how do you expect me to write that? Using latex only for the sum symbol, like in `$\sum_{i = 1}^{n - 1}$`? Using simililatex by writing `∑_{i = 1}^{n - 1}`? Or is what I had in the beginning fine enough?

Reply:
IMO LaTeX is readable enough if you have more than 0 spaces in your expression: `$\sum_{i=0}^0 \dfrac{a_0 a_{0+1}}{a_0 a_{0+1}} = \dfrac{a_0 a_1}{a_0 a_1} = 1$.` Or, even better for me, replace `\dfrac` with a `/`: `$\sum_{i=0}^0 (a_0 a_{0+1}) / (a_0 a_{0+1}) = (a_0 a_1) / (a_0 a_1) = 1$.`, which balances nicely between readability in LaTeX and in plain text.